src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -70,6 +70,18 @@ public void TestBackgroundWorkerBasic()
             }
         }
 
+        [Fact]
+        public void RunWorkerAsync_NoOnWorkHandler_SetsResultToNull()
+        {
+            var backgroundWorker = new BackgroundWorker { WorkerReportsProgress = true };
+            backgroundWorker.RunWorkerCompleted += (sender, e) =>
+            {
+                Assert.Null(e.Result);
+                Assert.False(backgroundWorker.IsBusy);
+            };
+            backgroundWorker.RunWorkerAsync();
```

Comment:
Interesting. I've updated the test to handle this fact, and also to make sure that the completed handler was called